### ISSUE TITLE
Use the app grid icon instead of the ubuntu logo

### ISSUE THREAD

**Issue by Feichtmeier** [ubuntu/communitheme#584]
Edit: it has been decided that this theme will follow upstream and use an app grid icon for the "show apps" button in gnome shell/dash to dock, so we will discuss if and how we can improve the icon regarding our theme

Previously discussed here:
https://github.com/ubuntu/gnome-shell-communitheme/pull/137
https://github.com/ubuntu/gnome-shell-communitheme/issues/134

**Comment by eaglersdeveloper** [comment]
I suggest use the icon in Suru style:

![menu_icon](https://user-images.githubusercontent.com/40228953/42128535-00c6552c-7cbf-11e8-9c2f-e46e937a562f.png)

I took idea from the Scopes icon:

![](http://ubuntu-news.ru/sites/default/files/unity-8-desktop-spread-750x563.png)

**Comment by galgalesh** [comment]
I think in terms of usability, you can argue either way. Using "OS logo" for "applications menu" is common:

* Unity7 uses the Ubuntu logo for the applications dash
* Windows uses the Windows logo for the start menu

**Comment by ya-d** [comment]
Sorry, but I'll [quote myself](https://github.com/ubuntu/gnome-shell-communitheme/pull/137#issuecomment-385210956) here. Even in Unity the dash is for much more than apps, its a *system wide* function to discover files or media, look up dictionaries *plus* open apps.

> From a usability/UX standpoint I think this is quite a bit different to macOS or Windows. macOS uses the Apple logo for it's system menu (About, Settings, App Store, Power, …), Windows uses its own logo for the "start" menu (Files, Settings, Power + Apps). While "Show Applications" is, well, just that: an app overview. So while the brand logo could be related to "the system" in general, I fail to see the relation of the Ubuntu logo in Communitheme to anything Ubuntu.

**Comment by Feichtmeier** [comment]
Na it's not only an app overview. You can type and search also from there :)
The GNOME Dash has 3 states
- window spread + workspaces
- app grid
- search view

You can go into the search view from either one of the two.

**Comment by ya-d** [comment]
Search is more or less a filter for which applications are shown (maybe Settings are added in, not sure). Activities (windows + workspaces) seems much more like what "the system" or OS does to me, I have no real gripe with adding an Ubuntu logo there.

**Comment by Feichtmeier** [comment]
Well there are several arguments pro and con the ubuntu logo as the app grid logo. 
Here are two pros:
- the ubuntu logo is used in unity for the dash, where applications are shown and you can filter them by clicking filters or by searching
- the ubuntu logo used as the app grid icon could be a heavy branding... imagine the guides ;D "Press the Ubuntu-Button"

The con's are in the opening post. But either way, we have to wait for guadec and didis return with the final decision. If GNOME doesn't like this branding we use the app grid icon.

**Comment by galgalesh** [comment]
Just to clarify something to @ya-d 

Regardless of how you open search in Gnome 3; it will show all search providers. This includes

* Gnome calculator solving equations you type in the search box
* Gnome files showing files matching the search
* Gnome Software showing not-installed applications matching the search
* Gnome Settings showing settings matching your search

This is almost exactly the behavior of Unity and Windows.

However, I agree to go with the recommendations of the Gnome folks.

**Comment by ya-d** [comment]
Oh, ok. Thanks for correcting that, Merlijn! I'll stick with the difference to Windows and Mac though as they provide system functions like shutdown etc in their branded menus.

But I'm not saying there are no other arguments by any means, I just wanted to add mine to the conversation. There are pretty reasonable points either way which have to be weighted. I'm sure you'll find a good solution for branding and usability while still pleasing upstream.

*??  Maybe there could be same kind of a "badge" to the icon: either an app grid icon with an Ubuntu badge or an Ubuntu logo with an app grid badge. Just a quick thought. Probably too much/busy.*

**Comment by godlyranchdressing** [comment]
I really like the Suru icon idea, the current monochrome logo looks a little unfriendly.

**Comment by madsrh** [comment]
> I really like the Suru icon idea, the current monochrome logo looks a little unfriendly.

Really? I really the the current monochrome very much (as mentioned elsewhere 🤷‍♂️ )

The Suru icon, looks exactly like that... an icon from an application. I think it's kind of misleading

![image](https://user-images.githubusercontent.com/3986894/42424983-2e2127a6-8316-11e8-8afc-5c29ccdf607e.png)

**Comment by godlyranchdressing** [comment]
@madsrh I agree about that too, but still, I like it! :laughing: 

Reposting a mockup from @taciturasa on the community forum.

![90ff746db0f8e8c3219ff91a62ca08d0907c76c2_1_690x387](https://user-images.githubusercontent.com/27529229/42453237-1b768d6e-835a-11e8-84ea-6dda7a8b2b17.png)

I know we're still waiting on @didrocks's response on replacing upstream stuff.

**Comment by didrocks** [comment]
So, we just discussed this as part of GUADEC. The general agreement is that it doesn't make sense to replace the app grid with the ubuntu logo.

We can explore adding it on top, next to Activities, but keep the "activities" wording please. Sounds good for everyone?

**Comment by Feichtmeier** [comment]
@didrocks great news! Then we have at least some ubuntu logo - and in the end we all agreed that the app grid button made more sense.
@godlyranchdressing could you make a PR (again) for this one? :)

**Comment by galgalesh** [comment]
@didrocks I'm curious to hear the reasoning behind this. I don't mind going with their guidelines, but I'd like to hear their explanation because to me this seems a counter-intuitive decision; activities is similar to a "window spread" on Unity and "task view" on Windows. It's not a menu so it seems to go against this common UI pattern. :thinking:

From a UX perspective, the best thing might be to put the "Task View" icon from Windows (![image](https://user-images.githubusercontent.com/1492981/42464748-48ba6b60-83aa-11e8-8eb9-4e90a38a6bc8.png)) next to activities. But then we don't have any branding which is _always_ visible.. Bye bye Ubuntu logo on a hacker's laptop in movies.. :smile:


*Note: To add to the confusion, earlier versions of Unity actually had a very similar button for the dash(menu)*

![image](https://user-images.githubusercontent.com/1492981/42464397-2d5b5ee8-83a9-11e8-9fdf-4e0196d5d0c0.png)

**Comment by taciturasa** [comment]
^ This. I understand it's not the way it is in GNOME, so there's that confusion, but we're already moving away from a lot of default GNOME patterns, esp. the dock, so I don't see why it can't be changed for the better here. I've outlined my reasoning in the communitheme thread on discourse, with even more reasoning as to why it should maybe be an upstream thing too. I'm not too torn either way, as all I've done so far is mess around in inkscape/gimp a bit, but it is a bit frustrating that the reasoning seems so vague.

Just to pop this down from there, for reference:

>The change from a label to an OS button may seem weird, but, paradoxically, I feel it may be more communicative to new and old users what that button does than “Activities” does. “Activities” doesn’t relay the idea of the gnome dash very well, but users have been trained on what a start menu is with Windows. It’ll be a likely thing for a new user to explore, and it’ll communicate the space as incredibly important to the experience like it should be: just like Windows Start or macOS Apple menu, the branding is a signpost.
>
>That, and I feel it’s pill shape would make for a very good home button/modern gesture slider for GNOME Mobile, when purism’s new mobile shell hits critical mass and if ubuntu ever makes its way onto that :wink:

and later:

>It makes maybe more sense in the context of vanilla GNOME; there, it really is the one main menu for everything, so it makes sense for it to be branded like the start button (on vanilla GNOME, that being the GNOME footprint), as well as giving the shell a strong identity for users. That’s where you “start” when you turn your computer on; it’s like a table where everything is laid out for you. It combines the start menu and the task view, thinking in terms of windows. It’s the GNOME way.
>
>It also makes sense in respect to Purism’s GNOME Mobile efforts: instead of an activities label, they’re using an up arrow to signify the “home” (which for now is just the app grid, but in the future will be the full gnome dash, if i read right). A pill works better than that, imo, as not only is it a more regonizable home button, it also opens the door to future gesture control the likes of which we’re seeing in iOS, Android, Moto phones, etc., that actually came from webOS, GNOME’s direct influence. It also just works visually on both desktop and mobile devices.
>
>That’s not to say that it doesn’t still make sense in regards to ubuntu; the gnome dash is still the most important paradigm in GNOME, and basically what the whole experience is centered around. It’s still the “main table”. Using the ubuntu logo instead of the gnome footprint would make more sense for this instance, but it still serves most of the same purpose.

**Comment by Feichtmeier** [comment]
![screenshot from 2018-07-09 19-19-04](https://user-images.githubusercontent.com/15329494/42465711-0d5fe056-83ad-11e8-8b30-b21577ee2b23.png)

That's the best I've made so far... I hope Aaron comes online soon x D

_common.scss:
#panelActivities.panel-button {
    background-image: url("activities.svg");
    background-size: contain;
    text-align: right;
    width: 100px;
}

_dock.scss:
  .show-apps {
    .overview-icon {
      border-radius: $small_radius;
    }

    &:active, &:checked {
      .overview-icon { background-color: $base_active_color; box-shadow: none; }
    }
  }

+ a custom .svg file I've created

**Comment by taciturasa** [comment]
As a quick addendum before I step away for a bit, activities would great as
a tool too for the button, though I believe that'd require upstream
changes.

**Comment by Feichtmeier** [comment]
@galgalesh I was thinking about this, too.

The ubuntu button for activities makes as much sense as the ubuntu button for the app grid. :man_shrugging: 
In both cases I would just have liked it because of the "ubuntu-branding".

**Comment by taciturasa** [comment]
>Although I just realized that if you press the "Ubuntu" button on your laptop, you open activities instead of the app grid 

This is something I was thinking too with the pill. It's basically your home button, no matter the form factor.

**Comment by taciturasa** [comment]
@Feichtmeier right, not trying to be too emotional about this one way or the other. Sorry if that's the way it came off!

**Comment by taciturasa** [comment]
@Feichtmeier right, not trying to be too emotional about this one way or the other. Sorry if that's the way it came off!

**Comment by Feichtmeier** [comment]
@taciturasa np, this whole topic is kind of hot! :dancer: :smile: 

![activities-p1ci02cp5i1g8iquvss11aqg](https://user-images.githubusercontent.com/15329494/42467070-1fdf63d8-83b1-11e8-95ad-eb80fd857384.gif)

```sass
#panelActivities {
    background-image: url("activities.svg");
    background-size: contain;
    text-align: right;
    width: 110px;
    height: 24px;
}```

After making the length smaller we got a better spacing. Yet the icon has the strange jumping effect :thinking:

**Comment by taciturasa** [comment]
@Feichtmeier looks great! Is it perhaps the box shadow for active that's messing things up?

Ignoring the other stuff (I didn't actually realize it was upstream GNOME designers that made the decision; I thought there was a third middleman party who did it), I wanna draw attention to the Suru-shaped app grid button I did, and see if anyone finds it a better fit than the default GNOME app grid button? It uses the GNOME app grid shape, just placed inside a white Suru squircle.

**Comment by Feichtmeier** [comment]
@taciturasa I liked the idea of drawing the shape around it. 
However, as @madsrh already pointed out, if it has the same "paper-effects" like box.shadows and outlines, it looks like it's some application and not part of the shell functionality. But I couldn't tell from the screenshot you've attached if it has those effects or if it is flat.

**Comment by taciturasa** [comment]
Nope, it indeed doesn't have the paper effects - that was something I was cognizant of 😄 

Here's the full size of it, so you can see for yourself. 
![buttontest](https://user-images.githubusercontent.com/18127936/42467918-4c6fefd8-8381-11e8-99a2-1c09d1eba811.png)

**Comment by Feichtmeier** [comment]
@taciturasa could you give me a link to the file? Please .SVG

**Comment by taciturasa** [comment]
It'll be ~7 hours until I get home and can grab the svg, but I'll plop it
down as soon as I can!

**Comment by taciturasa** [comment]
Okay, next question is can we play around with the Ubuntu logo plus
activities wording INSIDE the pill? Sorry for still keeping this going, I'm
a lame and unabashed webOS apologies :P

**Comment by Feichtmeier** [comment]
I also thought about this, since I liked your mock-up. However, this would reduce the customization since you would need to place the text inside the svg.

**Comment by Feichtmeier** [comment]
Read what didier reported from guadec. Ubuntu logo yes but with the text :)

**Comment by taciturasa** [comment]
Sorry, miscommunication - I meant in regards to the Ubuntu logo next to the
text :)

**Comment by didrocks** [comment]
I guess the whole discussion should be on the hub anyway, to include more community members. I just wanted to update on that PR here as it was related ;)

We are going to discuss general themability this afternoon at GUADEC btw, not only this icon where we took the decision yesterday about it.
In a nutshell, the grid icon shows a grid. Putting the grid icon as the top (as stated multiple times in my blog post suit) isn't suitable, because you would have 2 similar buttons spacially closed, (Activity and app grid), having slightly different behavior.

**Comment by taciturasa** [comment]
Again think that was another miscommunication. Feicht was adding an Ubuntu
logo next to the activities text, and I was wondering if it wasn't a better
idea to go with a white Ubuntu logo instead of the one in the circle that
he has now.

**Comment by Feichtmeier** [comment]
@taciturasa please post your new grid icon ideas in here. I've merged the PR for now because we need to get a clean gnome shell repo because didier will move the code into this repo (and change the build settings ofc).

**Comment by taciturasa** [comment]
Gotcha!

**Comment by eaglersdeveloper** [comment]
Let's look at the various Android shells.
Despite the fact that in Android Nougat and higher renounce app drawer, 
there are still shells in which you can optionally turn it on.

Samsung Experience (optional):

Where is the apps button on my Home screen? How do I find all my apps?


LG UX (optional):


Huawei EMUI (optional):

emui_5_with_app_drawer_enabled


Sony Xperia UI (default):

Картинки по запросу sony xperia ui


HTC Sense (default):

Картинки по запросу htc sense 9 screenshot

Almost everyone uses app grid in shape. Everything except HTC Sense.

12.07.2018 20:50, taciturasa пишет:

**Comment by eaglersdeveloper** [comment]
Let's look at the various Android shells.
Despite the fact that in Android Nougat and higher renounce app drawer, there are still shells in which you can optionally turn it on.

Samsung Experience (optional):
![](https://images.samsung.com/is/image/samsung/p5/uk/support/mobile-devices/where-is-the-apps-button-on-my-home-screen-how-do-i-find-all-my-apps/images/where-is-the-apps-button-on-my-home-screen-how-do-i-find-all-my-apps-6.png?$ORIGIN_PNG$)

LG UX (optional):
![](http://cs5-3.4pda.to/7876694.jpg)

Huawei EMUI (optional):
![](https://img.talkandroid.com/uploads/2016/12/EMUI_5_with_App_drawer_enabled-338x600.jpg)

Sony Xperia UI (default):
![](https://cdn57.androidauthority.net/wp-content/uploads/2016/10/Sony-Xperia-XZ-screenshots-7-300x533.jpg)

HTC Sense (default):
![](https://cdn.wccftech.com/wp-content/uploads/2015/03/HTC-Sense73-416x740.png)

Almost everyone uses app grid in shape. Everything except HTC Sense.

**Comment by Feichtmeier** [comment]
There might be a reason to follow a different desktop environment but:
- Ubuntu in gnome DE has a background already (the dock!) and thus a background on a background doesn't seem that logical anymore as you might think in the first moment
- there are also Android launchers which hide it completely and work with only swiping up

**Comment by eaglersdeveloper** [comment]
> there are also Android launchers which hide it completely and work with only swiping up

I talked about this:
>  Despite the fact that in Android Nougat and higher renounce app drawer, there are still shells in which you can optionally turn it on.

In Pixel Launcher and Samsung Experience the menu opens with swipe up.
In LG UX, MIUI, EMUI, Flyme and other Asian shells apps located on the homescreen by default.

Why are we talking about mobile OS? Let's go back to the desktop!

Chrome OS. The app menu icon resembles the Android's Home icon. Background of this icon: a translucent circle, looks like circle application icons (if not visible, enlarge image).
![](https://www.androidpolice.com/wp-content/uploads/2018/02/nexus2cee_chrome-os-pwa.png)

Windows 10 Taskbar. Start is button with monochrome Windows logo. But icons of PWA apps also monochrome.
![](https://mk0intowindows84fvao.kinstacdn.com/wp-content/uploads/2015/09/Customize-Windows-10-taskbar-pic7_thumb.png)

macOS Dock. Launchpad is responsible for showing the list of applications. And its icon app grid or a system logo. It's a rocket.
![](http://cdn.osxdaily.com/wp-content/uploads/2014/10/yosemite-flat-dock.jpg).
The application with the macOS logo is the file manager. However, it provides a shell, as Windows Explorer.

**Comment by Feichtmeier** [comment]
?
I don't know, you brought the android screenshots in here.

The point is it will be an appgrid icon that's for sure. No matter if Windows has Windows button for their windows things or macos got that macos button.

Upstream has an appgrid and  this theme will have an appgrid too.

So for whatever reason the upstream icon *might* not fit into our monochrome icon set we use, we could search for a slightly alternated look. 
So I am still looking forward for tracituras icon.

**Comment by eaglersdeveloper** [comment]
> I don't know, you brought the android screenshots in here.

I showed that designers rarely use the app grid with a shape.


13.07.2018 14:17, Feichtmeier пишет:

**Comment by taciturasa** [comment]
I don't exactly see how that's relevant. That attempt was ruled out. Usability-wise, it's nothing but stylistic preference. It's not easier or harder to use a bare app grid or an app grid within a shape on any form factor.

**Comment by Luxamman** [comment]
The app grid icon is not right for me to be honest. 
Have to say that the logo in the top-bar isn't working for me either (and the logo branding is gone anyway because it looks just like a circle) and already miss the nice Ubuntu (branding) logo as the app launcher. 

We discussed that several times and so I'm a little confused why to shake up everything again. Maybe we can have a kind of activities icon as a replacement for "Activities" (like a multi-window icon) but it needs to be the same width like the dock underneath to look good. But this also can make problems with the upstream...?

**Comment by Feichtmeier** [comment]
The reason for the appgrid is the news Didi brought from guadec (they don't want anything else than the app grid logo there because it's the appgrid button)

**Comment by Luxamman** [comment]
Well. Disappointing.

**Comment by Paz-it** [comment]
So it can be a tiny appgrid ::: surrounded by the circle of friends logo.  :thinking:

**Comment by Feichtmeier** [comment]
My problem would be not the fact that there is the appgrid icon but that I need to fiddle with the Ubuntu logo now at activities. Because it always is wrong aligned in any of the situations (long string or short string).

**Comment by Feichtmeier** [comment]
![peek 2018-07-16 10-58](https://user-images.githubusercontent.com/15329494/42750674-5bca23e2-88e8-11e8-83b6-98bcde4f3aed.gif)
We could also try "something like this"
If we would use a clever combination of forms there could be a nice transition effect.
However we would need to double check if this would be against upstream yet again

**Comment by taciturasa** [comment]
https://owo.whats-th.is/59be37.svg

Here's a quick first test, sorry it took so long; been busy. Think the squares might need to be slightly smaller, as well as have less border radius.

**Comment by Feichtmeier** [comment]
@taciturasa thanks for your work.

I must admit that I'd rather use your first icon or the upstream one. This one looks really big 
![screenshot from 2018-07-16 23-37-04](https://user-images.githubusercontent.com/15329494/42785078-642f9b20-8951-11e8-81ab-0d7e011270f6.png)

**Comment by taciturasa** [comment]
Yeah, that's what I was afraid of. More attempts on the way.

**Comment by Feichtmeier** [comment]
https://github.com/ubuntu/gnome-shell-communitheme/pull/243

Going to remove the logo in the activities button again, we have a lot of issues because of this. I shouldn't have merged it in the first place.
Going with your 2. icon for now (transparent grid)

**Comment by Luxamman** [comment]
The actually used icon with the white background looks okay to me and kind of fits, but still funny that the community wants to have more branding than Ubuntu or GNOME...

**Comment by Feichtmeier** [comment]
I think this comes the fact that many users who try the theme have a unity 7 background.

**Comment by taciturasa** [comment]
https://owo.whats-th.is/270c31.svg 'nother slight test

**Comment by Feichtmeier** [comment]
I am actually starting to get used to your squircle (but with the transparent holes) :laughing: 

I will try it later, thanks again for your work!

**Comment by taciturasa** [comment]
Oh, I can still go for that if you like it. I'd just use diferrent shapes to do the transparent holes, as well as adhere to the suru grid!

**Comment by taciturasa** [comment]
https://owo.whats-th.is/cee873.svg

Beep boop, wheep woop

**Comment by Feichtmeier** [comment]
This one looks really good: https://github.com/ubuntu/communitheme/issues/584#issuecomment-406000803
![screenshot from 2018-07-20 09-42-37](https://user-images.githubusercontent.com/15329494/43001017-fd4787e4-8c23-11e8-9a58-a3214af0c16f.png)

This one does not look good: https://github.com/ubuntu/communitheme/issues/584#issuecomment-406311923
![screenshot from 2018-07-20 13-47-37](https://user-images.githubusercontent.com/15329494/43001029-0d873ef6-8c24-11e8-97cb-4e66435c82cc.png)

**Comment by Feichtmeier** [comment]
Could you try to make a second one of the one "i like" but with the outer squircles moved a bit more to the middle. Then I could use this svg for when the button is pressed (just an idea). For this state:
![screenshot from 2018-07-20 13-56-20](https://user-images.githubusercontent.com/15329494/43001251-e4753dd2-8c24-11e8-9717-57b7983db06e.png)

**Comment by 92Julian** [comment]
> it is ofc really sad to see an Ubuntu desktop without the Ubuntu logo.

Why not use <a href="https://extensions.gnome.org/extension/889/background-logo/">Background Logo</a>?
![imagen](https://user-images.githubusercontent.com/25293942/43018259-78074d56-8c2f-11e8-8ac9-ee505cea4e13.png)

**Comment by taciturasa** [comment]
Hey, so, I talked to snwh, who said that the icon shape should NOT be used for non icons. this means that the shape probably needs to be reverted back to upstream until better ones emerge.

**Comment by Feichtmeier** [comment]
Yeah I had similar thoughts about this in the beginning but after some weeks I've started to like the grid! ;D
However, his argument is pretty solid.

Anyways, you also put one without the background which we could use. I even got a branch for this somewhere :thinking:

**Comment by taciturasa** [comment]
Please don't use the one without the background, it's really bad :sweat: At
the very least, make the dots on it smaller.

**Comment by Feichtmeier** [comment]
Well if even you don't want it anymore, we can just go full upstream and close this issue - finally?

**Comment by taciturasa** [comment]
Sure, sounds fine to me.

**Comment by eaglersdeveloper** [comment]
Upstream app grid icon looks too classic and old.
How about the round icon like in Android 5.0-7.0?

![2018-07-23 19-08-36](https://user-images.githubusercontent.com/40228953/43085426-0dfb57c0-8eac-11e8-97a6-e2a7957aac06.png)

**Comment by Feichtmeier** [comment]
> So, we just discussed this as part of GUADEC. The general agreement is that it doesn't make sense to replace the app grid with the ubuntu logo.
> 
> We can explore adding it on top, next to Activities, but keep the "activities" wording please. Sounds good for everyone?

Ah, found it. My necro skills emerge

@didrocks do you still think that we need to be sensitive with "Activities" ? I actually like this idea to replace it with the ubuntu pill @Austinuity

**Comment by didrocks** [comment]
I think this is a discussion that should include as well the design team and takes a decision there.
I didn’t see anything that changed in the Shell design which would modify this statement and agreement with GNOME design team, but maybe there is?

**Comment by Feichtmeier** [comment]
> I think this is a discussion that should include as well the design team and takes a decision there.
> I didn’t see anything that changed in the Shell design which would modify this statement and agreement with GNOME design team, but maybe there is?

There is a discussion happening on gnome gitlab about replacing activities with an icon.
Sure we could wait until this all happened upstream. Alternatively if we see the wind blowing in this direction we could eventually use the CoF pill already 0:)

**Comment by didrocks** [comment]
:+1:
